Spanner/tests/Unit/DatabaseTest.php

```diff
@@ -1452,4 +1475,80 @@ public function testDBDatabaseRole()
         $sql = 'SELECT * FROM Table';
         $this->databaseWithDatabaseRole->execute($sql);
     }
+
+    public function testExecuteWithDirectedReadOptions()
+    {
+        $this->connection->executeStreamingSql(Argument::withEntry(
+            'directedReadOptions',
+            $this->directedReadOptionsIncludeReplicas
+        ))
+        ->shouldBeCalled()
+        ->willReturn($this->resultGenerator());
+
+        $sql = 'SELECT * FROM Table';
+        $res = $this->database->execute($sql);
+        $this->assertInstanceOf(Result::class, $res);
+        $rows = iterator_to_array($res->rows());
+    }
+
+    public function testPrioritizeExecuteDirectedReadOptions()
+    {
+        $this->connection->executeStreamingSql(Argument::withEntry(
+            'directedReadOptions',
+            $this->directedReadOptionsExcludeReplicas
+        ))
+        ->shouldBeCalled()
+        ->willReturn($this->resultGenerator());
+
+        $sql = 'SELECT * FROM Table';
+        $res = $this->database->execute(
+            $sql,
+            ['directedReadOptions' => $this->directedReadOptionsExcludeReplicas]
+        );
+        $this->assertInstanceOf(Result::class, $res);
+        $rows = iterator_to_array($res->rows());
```

Comment:
Fetching of rows should be followed by an assertion on the fetched value, recommending to add an assertion on resulting row count.
`$this->assertCount(<some magic number>, $rows);`

This is also repeated below